etools-upload.js

```diff
@@ -377,6 +380,18 @@ class EtoolsUpload extends RequestHelperMixin(CommonMixin(PolymerElement)) {
       this.resetValidations();
     }
   }
+
+  validFileType(fileName) {
+    const acceptedExtensions = this.accept.split(',');
+    const fileExtension = fileName.slice((fileName.lastIndexOf(".") - 1 >>> 0) + 2);
```

Comment:
Can you please explain this code? :
`(fileName.lastIndexOf(".") - 1 >>> 0) + 2`